pkg/utils/utils.go

```diff
@@ -445,3 +456,189 @@ func DoesMigrationScheduleMigrateNamespaces(migrationSchedule stork_api.Migratio
+	securityContext := pod.Spec.SecurityContext
+	uid, gid = getIdFromSecurityContext(securityContext)
+	if uid != UndefinedId || gid != UndefinedId {
+		return uid, gid, nil
+	}
+	// Neither pod nor container has the securityContext defined with User Id or group Id, return error with undefined uids
+	return uid, gid, fmt.Errorf("failed to get user id or group id from securityContext of pod [%s/%s]", pod.Namespace, pod.Name)
+}
+
+// Get uid gid from application backup cr
+func GetUIDGIDFromBackupCR(backup *stork_api.ApplicationBackup, pvcName string) (int64, int64, error) {
+	fn := "GetUIDGIDFromBackupCR"
+	// read the uid and gid from the application backup cr's volume array
+	if backup != nil {
+		for _, volume := range backup.Status.Volumes {
+			if volume.PersistentVolumeClaim == pvcName {
+				logrus.Infof("%v: Found uid:[%v] and gid:[%v] for the restore of the pvc[%v]", fn, volume.VolumeJobSecurityContext.RunAsUser, volume.VolumeJobSecurityContext.RunAsGroup, volume.PersistentVolumeClaim)
+				return volume.VolumeJobSecurityContext.RunAsUser, volume.VolumeJobSecurityContext.RunAsGroup, nil
+			}
+		}
+	}
+	err := fmt.Errorf("failed to obtain uid from backup cr")
+	return UndefinedId, UndefinedId, err
+}
+
+// GetPsaEnabledAppUID - Get the UID/GID of the application pod which has psa enforcement in its namespace using the PVC
+func GetPsaEnabledAppUID(pvcName string, namespace string, backup *stork_api.ApplicationBackup, getUIDFromApp bool) (int64, int64, bool, error) {
+	fn := "GetPsaEnabledAppUID"
+	uid := UndefinedId
+	gid := UndefinedId
+	psaIsEnforced := false
+
+	// Check if the namespace has the PSA labels set
+	psaIsEnforced, psaLabelMap, err := GetPsaDetail(namespace)
+	if err != nil {
+		logrus.Errorf("%s: %v", fn, err)
+		return uid, gid, psaIsEnforced, err
+	}
+	if !psaIsEnforced {
+		errMsg := fmt.Sprintf("PSA is not enforced in namespace [%s], no need to enforce any uid/gid in job for backing up pvc [%s]", namespace, pvcName)
+		logrus.Debugf("%s: %v", fn, errMsg)
+		return uid, gid, psaIsEnforced, fmt.Errorf(errMsg)
+	}
+
+	// check in the psa label map if the pod has the psa enforcement labels set to restricted
+	for k, v := range psaLabelMap {
+		// Only in case of restricted enforcement, we need to get the UID/GID of the pod
+		// otherwise baseline and privilege will work if the job pod runs with root access.
+		// TODO: Should we preserve the UID in case baseline config too, though not required for JOB.
+		if strings.Contains(k, "enforce") && v == "restricted" {
+			if !getUIDFromApp {
+				//get uid gid from applicationBackup CR for restore
+				uid, gid, err = GetUIDGIDFromBackupCR(backup, pvcName)
+				if err != nil {
+					logrus.Errorf("%s: %v", fn, err)
+					return uid, gid, psaIsEnforced, err
+				}
+			} else {
+				// Get the pod which uses this PVC for backup scenario
+				pod, err := GetPodFromPVC(pvcName, namespace)
```

Comment:
As we are not going to support PVC's which are lying around and not used by any Pods, i prefer we return appripriate error message from here and in the caller check for that and move onto next PVC and in future we will mark this as PartialSuccess. 
With the current change when you handle error of returning from `GetPsaEnabledAppUID` this type of backup combination would lead to 30 min timeout.